lib/elixir/test/elixir/kernel/errors_test.exs

```diff
@@ -719,6 +719,30 @@ defmodule Kernel.ErrorsTest do
     end
   end
 
+  test "@on_load attribute with undefined function" do
+    assert_eval_raise CompileError,
+                      "nofile:1: @on_load function foo/0 is undefined",
+                      'defmodule UndefinedOnLoadAttribute do @on_load :foo end'
+  end
+
+  test "wrong kind for @on_load attribute" do
+    # Capture warning: function foo/0 is unused
+    ExUnit.CaptureIO.capture_io(:stderr, fn ->
```

Comment:
If we do this, the testcase cannot be async. So let's define a public function that calls `defp foo` so we don't need to capture the warning.

Reply:
Good catch! I completely forgot about that!

Reply:
Done in https://github.com/elixir-lang/elixir/pull/7973/commits/86c46d744f560c8f3f785c6c0788174778d5474b